src/sparkx/Oscar.py

```diff
@@ -306,7 +307,9 @@ def print_particle_lists_to_file(self, output_file: str) -> None:
                 raise ValueError("The number of output per event is empty.")
             if self.num_events_ is None:
                 raise ValueError("The number of events is empty.")
-            if self.num_events_ > 1:
+            if self.num_events_ == 0:
```

Comment:
You were reporting issues with the case that there are no events left - is a warning sufficient to deal with this? 
Also, such changes have to be done always for both oscar and Jetscape. You would have easily noticed that running the tests just once.